spec/lib/baw-workers/analysis/payload_spec.rb

```diff
@@ -1,12 +1,159 @@
   }
 
+  let(:analysis_params) {
+    {
+        command_format: '<{file_executable}> "analysis_type -source <{file_source}> -config <{file_config}> -output <{dir_output}> -tempdir <{dir_temp}>"',
+        config: 'blah a very long and repeated blob of text' * 300,
+        file_executable: 'echo',
+        copy_paths: [],
 
+        uuid: 'f7229504-76c5-4f88-90fc-b7c3f5a8732e',
+        id: 123456,
+        datetime_with_offset: '2014-11-18T16:05:00Z',
+        original_format: 'wav',
 
+        job_id: 20,
+        sub_folders: ['hello', 'here_i_am']
+    }
+  }
+
+  let(:analysis_params2) {
+    {
+        command_format: '<{file_executable}> "analysis_type -source <{file_source}> -config <{file_config}> -output <{dir_output}> -tempdir <{dir_temp}>"',
+        config: 'blah a very long and repeated blob of text' * 300,
+        file_executable: 'echo',
+        copy_paths: [],
+
+        uuid: '146d4563-18e7-4aea-ab46-bce5c69dc68c',
+        id: 654321,
+        datetime_with_offset: '2014-11-19T16:05:00Z',
+        original_format: 'wav',
+
+        job_id: 20,
+        sub_folders: ['hello', 'here_i_am']
+    }
+  }
+
+  let(:expected_payload) {
+    {
+        "class" => "BawWorkers::Analysis::Action",
+        "args" => [
+            '7f3600c1257e4a334f759d9f4de2629e',
+            {
+                "analysis_params" =>
+                    {
+                        'payload_base' => 'baw-workers:partial_payload:analysis:20-1471651200',
+                        "uuid" => "f7229504-76c5-4f88-90fc-b7c3f5a8732e",
+                        "id" => 123456,
+                        "datetime_with_offset" => "2014-11-18T16:05:00Z",
+                        "original_format" => "wav",
+                        "job_id" => 20,
+                    }
+            }
+        ]
+    }
+  }
+
+  let(:expected_payload2) {
+    {
+        "class" => "BawWorkers::Analysis::Action",
+        "args" => [
+            '9bbcedbfa382f507a8f9209500dd4486',
+            {
+                "analysis_params" =>
+                    {
+                        'payload_base' => 'baw-workers:partial_payload:analysis:20-1471651200',
+                        "uuid" => "146d4563-18e7-4aea-ab46-bce5c69dc68c",
+                        "id" => 654321,
+                        "datetime_with_offset" => "2014-11-19T16:05:00Z",
+                        "original_format" => "wav",
+                        "job_id" => 20,
+                    }
+            }
+        ]
+    }
+  }
+
+  let(:partial_payload) {
+    {
+
+        "command_format" =>
+            "<{file_executable}> \"analysis_type -source <{file_source}> -config <{file_config}> -output <{dir_output}> -tempdir <{dir_temp}>\"",
+        "sub_folders" => ['hello', 'here_i_am'],
+        "config" => 'blah a very long and repeated blob of text' * 300,
+        "file_executable" => "echo",
+        "copy_paths" => []
+
+    }
+  }
+
+
+  it 'stores and retrieves partial payloads' do
```

Comment:
Yes, 2.3.1.

Error is on the check comparing the size of the found1 array. https://github.com/QutBioacoustics/baw-workers/pull/42/files#diff-fd2c3b29f76f9da90fd4e4832383c710R123

Not sure what's up, could be an issue with the order of args...